ledger/indexer-benchmark/src/main/scala/ledger/indexerbenchmark/IndexerBenchmark.scala

```diff
@@ -0,0 +1,227 @@
+    PostgresResource
+      .owner()
+      .use(db => {
+        println(s"Running the indexer benchmark against the ephemeral Postgres database ${db.url}")
+        run(createUpdates, config.copy(indexerConfig = config.indexerConfig.copy(jdbcUrl = db.url)))
+        println(s"Run finished")
+        Future.unit
+      })
+    ()
+  }
+
+  def run(
+      createUpdates: String => Future[Iterator[(Offset, Update)]],
+      config: Config,
+  ): Unit = {
+    val metricRegistry = new MetricRegistry
+    val metrics = new Metrics(metricRegistry)
+    metrics.registry.registerAll(new JvmMetricSet)
+
+    val system = ActorSystem("IndexerBenchmark")
+    implicit val materializer: Materializer = Materializer(system)
+    implicit val resourceContext: ResourceContext = ResourceContext(system.dispatcher)
+
+    val indexerE = Executors.newWorkStealingPool()
+    val indexerEC = ExecutionContext.fromExecutor(indexerE)
+
+    newLoggingContext { implicit loggingContext =>
+      println("Generating state updates...")
+      val updates = Await.result(createUpdates(config.updateSource), Duration(5, "minute"))
+
+      println("Creating read service and indexer...")
+      val readService = createReadService(updates, config)
+      val indexerFactory = new JdbcIndexer.Factory(
+        ServerRole.Indexer,
+        config.indexerConfig,
+        readService,
+        indexerEC,
+        metrics,
+        LfValueTranslationCache.Cache.none,
+      )
+
+      val resource = for {
+        _ <- config.metricsReporter.fold(Resource.unit)(reporter =>
+          ResourceOwner
+            .forCloseable(() => reporter.register(metrics.registry))
+            .map(_.start(config.metricsReportingInterval.getSeconds, TimeUnit.SECONDS))
+            .acquire()
+        )
+
+        _ = println("Setting up the index database...")
+        indexer <- Await
+          .result(indexerFactory.migrateSchema(false), Duration(100, "hour"))
+          .acquire()
+
+        _ = println("Starting the indexing...")
+        startTime = System.nanoTime()
+        handle <- indexer.subscription(readService).acquire()
+
+        _ <- Resource.fromFuture(handle.completed())
+        stopTime = System.nanoTime()
+        _ = println("Indexing done.")
+      } yield {
+        val duration: Double = (stopTime - startTime).toDouble / 1000000000.0
+        val updates: Long = metrics.daml.parallelIndexer.updates.getCount
+        println(
+          s"""
+             |--------------------------------------------------------------------------------
+             |Indexer benchmark results
+             |--------------------------------------------------------------------------------
+             |
+             |Input:
+             |  source:   ${config.updateSource}
+             |  count:    ${config.updateCount}
+             |  jdbcUrl:  ${config.indexerConfig.jdbcUrl}
+             |
+             |Indexer parameters:
```

Comment:
We seem to be able to run this against any indexer implementation (which I like a lot), would be able somehow to render the indexer config with reflection, so we do not couple this output to a particular version of a particular implementation? (reflection maybe?)

Reply:
I have added a simple output of the entire indexerConfig using the default case class rendering.

Reply:
I think that is not particularly useful because it wont have the names of the properties, just the property, so we would need to count positions in order to understand (not to mention if config changed between counting positions :) )
But I am also ok with leaving it as it is right now, and we improve on this as we need it for other indexers :ok:

Reply:
actually I take this back entirely: we anyway have to roll our config, and since we need to change that anyway as things change, to adapt this is not that big of an overhead